api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -105,6 +107,16 @@ public CohortDefinition getAnyEncounterDuringPeriod() {
         return new MappedParametersCohortDefinition(cd, "onOrAfter", "startDate", "onOrBefore", "endDate");
     }
 
+    @DocumentedDefinition("codedObsSearchAdvanced")
+    public CohortDefinition getPatientsHavingObs() {
+        CodedObsCohortDefinition cd = new CodedObsCohortDefinition();
```

Comment:
@mseaton @djazayeri  Do you recommend the method name be changed to `getCodedObsSearchAdvanced` instead? 
Also, is there a cohort definition (besides `CodedObsCohortDefinition`) better suited for the specific case of making observations searches for patients with the specified parameters?
I don't really understand how the coded values apply in this case, with a better understanding of that, I can add it to what we have already like @djazayeri mentioned.